pkg/kubelego/kubelego.go

```
@@ -340,5 +349,46 @@ func (kl *KubeLego) paramsLego() error {
 	} else {
 		kl.legoWatchNamespace = watchNamespace
 	}
+
+	keyType := os.Getenv("LEGO_KEY_TYPE")
+	switch keyType {
+	case kubelego.KeyTypeRsa:
+		kl.legoKeyType = kubelego.KeyTypeRsa
+		break
+	case kubelego.KeyTypeEcc:
+		kl.legoKeyType = kubelego.KeyTypeEcc
+		break
+	default:
+		kl.legoKeyType = kubelego.KeyTypeEcc
+		break
+	}
+
+	keySize := os.Getenv("LEGO_KEY_SIZE")
+	if keyType == kubelego.KeyTypeEcc {
+		switch keySize {
+		case "224":
+		case "256":
+		case "384":
+		case "521":
+			ks, err := strconv.ParseInt(keySize, 10, 0)
+			if err != nil {
+				kl.legoKeySize = kubelego.EccKeySize
+			}
+
+			kl.legoKeySize = int(ks)
+			break
+		default:
+			kl.legoKeySize = kubelego.EccKeySize
+			break
+		}
+	} else if keyType == kubelego.KeyTypeRsa {
+		ks, err := strconv.ParseInt(keySize, 10, 0)
+		if err != nil {
+			kl.legoKeySize = kubelego.RsaKeySize
```

Again, can we return an error here instead of assuming defaults silently?